api/src/main/java/org/openmrs/module/fhir2/providers/r3/AllergyIntoleranceFhirResourceProvider.java

```diff
@@ -76,6 +84,39 @@ public AllergyIntolerance getAllergyIntoleranceById(@IdParam @NotNull IdType id)
 		        .map(Provenance30_40::convertProvenance).collect(Collectors.toList());
 	}
 	
+	@Create
+	@SuppressWarnings("unused")
+	public MethodOutcome creatAllergyIntolerance(@ResourceParam AllergyIntolerance allergyIntolerance) {
+		return FhirProviderUtils.buildCreate(
+		    allergyIntoleranceService.create(AllergyIntolerance30_40.convertAllergyIntolerance(allergyIntolerance)));
+	}
+	
+	@Update
+	@SuppressWarnings("unused")
+	public MethodOutcome updateAllergyIntolerance(@IdParam IdType id, @ResourceParam AllergyIntolerance allergyIntolerance) {
+		if (id == null || id.getIdPart() == null) {
+			throw new InvalidRequestException("id must be specified to update");
+		}
+		
+		allergyIntolerance.setId(id.getIdPart());
+		
+		return FhirProviderUtils.buildUpdate(allergyIntoleranceService.update(id.getIdPart(),
+		    AllergyIntolerance30_40.convertAllergyIntolerance(allergyIntolerance)));
+	}
+	
+	@Delete
+	@SuppressWarnings("unused")
+	public OperationOutcome deleteAllergyIntolerance(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.AllergyIntolerance allergyIntolerance = allergyIntoleranceService.delete(id.getIdPart());
+		if (allergyIntolerance == null) {
+			throw new ResourceNotFoundException("Could not find allergyIntolerance to delete with id " + id.getIdPart());
+		}
+		OperationOutcome retVal = new OperationOutcome();
+		retVal.setId(id.getIdPart());
+		retVal.getText().setDivAsString("Deleted successfully");
+		return retVal;
```

Comment:
Make use of util method introduced by #214 
```java
return FhirProviderUtils.buildDelete(allergyIntolerance);
```

Reply:
sure for the operationoutcome right ?